docs/DesignDocs/Heap_Page_Permissions.md

```diff
@@ -0,0 +1,36 @@
+Heap Page Permissions
+=====================
+
+The permission flags on heap pages for SGX enclaves are currently hard-coded to `SGX_SECINFO_REG | SGX_SECINFO_R | SGX_SECINFO_W` in [create.c](https://github.com/openenclave/openenclave/blob/43e5e1fca9a55196ac94d5079eb3e615c1b4d6e6/host/sgx/create.c#L123). Some applications, for instance SGX-LKL, require the executable flag `SGX_SECINFO_X` as well. Given this difference in requirements, the permission flags should be a (signing-time) configuration setting, i.e. it is measured and attested.
```

Comment:
Is this proposal specific to SGX or not?  If not, then great.  If it is specific to SGX, please update the filename and title on line 1.

Reply:
I think we should aim for a unified interface/API on all platforms, but we do need an SGX version relatively soon. I hope that the mechanisms on other platforms are similar enough to make this feasible, but I haven't investigated this in detail yet.